pagopa/io-app
# chore: [IOPID-1015] Biometric refinement

## Short description
This pr adds the logic that allows the app not to show the biometric activation request if it has already been activated in a previous login and adds, in the case of FACE_ID, the prompt requesting permissions upon activation.

## List of changes proposed in this pull request
- ts/sagas/startup/onboarding/biometric/checkAcknowledgedFingerprintSaga.ts: added check to see if it had been activated in the last login.
- [ts/screens/onboarding/biometric&securityChecks/FingerprintScreen.tsx , ts/screens/profile/SecurityScreen.tsx] : added permission request prompt in case of faceid.
- ts/store/reducers/index.ts: biometric preference persisted to get it also after the end of a session.
- ts/store/reducers/onboarding.ts: added reset of biometric control also after session expired.
- ts/store/reducers/persistedPreferences.ts: added cross session control.
- ts/utils/biometrics.ts: added function that handles the biometric activation.

## How to test
With a new installation, try activating biometrics during onboarding:
- If the biometric is not FaceID, it should activate (or not) the feature without prompting.
- If the biometric is FaceID, when the active button is pressed, it should present the request for permissions:
    - If you give permissions and the authentication fails, it should not proceed.
    - If you give permissions and the authentication is successful, it should proceed and enable the feature.
    - If you don't give permissions, it should proceed and not activate the feature.
    
After logout or session expired, if the biometric has been activated during last session, it should not ask you for activation again after login, unless this is done with a different account than the previous one.
